docs/pages/release_notes.md

```diff
@@ -22,6 +22,17 @@ This is a {{ site.pmd.release_type }} release.
 
 ### API Changes
 
+#### Deprecated APIs
+
+* Visitor decorators are now deprecated and will be removed in PMD 7.0.0. They were originally a way to write
+  composable visitors, used in the metrics framework, but they didn't prove cost-effective.
+  * In `net.sourceforge.pmd.lang.java.ast`: JavaParserDecoratedVisitor, JavaParserControllessVisitor,
+    JavaParserControllessVisitorAdapter, and JavaParserVisitorDecorator are deprecated with no intended replacement
+  * In `net.sourceforge.pmd.lang.java.metrics.impl`:
+    * CycloAssertAwareDecorator, CycloBaseVisitor, and CycloPathAwareDecorator are deprecated and are replaced with CycloVisitor
+    * NcssBaseVisitor and NcssCountImportsDecorator are deprecated and are replaced with NcssVisitor
+
```

Comment:
We should propose an alternative, e.g. "For implementing own visitors, a JavaParserVisitorAdapter should be subclassed directly".